packages/jest-config/src/readConfigFileAndSetRootDir.ts

```diff
@@ -94,9 +95,40 @@ const loadTSConfigFile = async (
     });
   } catch (e: any) {
     if (e.code === 'MODULE_NOT_FOUND') {
-      throw new Error(
-        `Jest: 'ts-node' is required for the TypeScript configuration files. Make sure it is installed\nError: ${e.message}`,
-      );
+      let esbuild: typeof build;
+
+      try {
+        esbuild = require('esbuild').build;
+      } catch (error: any) {
+        if (error.code === 'MODULE_NOT_FOUND') {
+          throw new Error(
+            `Jest: 'ts-node' or 'esbuild' is required for the TypeScript configuration files. Make sure it is installed\nError: ${e.message}`,
+          );
+        }
+
+        throw error;
+      }
+
+      const outfile = 'node_modules/jest-config/_jest.config.js';
```

Comment:
What do you think of writing this file to the OS tmp directory instead? I'm not too fussed, but just thinking about how tmp is typically used and that arguably this is assuming `node_modules/jest-config` will exist which maybe it won't (e.g. due to "hoisting").

Reply:
Yes, I can do this.